CMakeLists.txt

```diff
@@ -1,21 +1,30 @@
 cmake_minimum_required (VERSION 3.2)
-project(xgboost)
+
+include("cmake/HunterGate.cmake")
+HunterGate(
+  URL "https://github.com/ruslo/hunter/archive/v0.19.221.tar.gz"
+  SHA1 "dc138cc9d1b5badf01e62ab0849f73c82e8ffd1d"  
+)
+
+project(xgboost VERSION 0.7.0)
 include(cmake/Utils.cmake)
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/modules")
 find_package(OpenMP)
 
-set_default_configuration_release()
 msvc_use_static_runtime()
 
 # Options
 option(USE_CUDA  "Build with GPU acceleration") 
 option(USE_AVX  "Build with AVX instructions. May not produce identical results due to approximate math." OFF) 
-option(USE_NCCL "Build using NCCL for multi-GPU. Also requires USE_CUDA") 
-option(JVM_BINDINGS "Build JVM bindings" OFF)
-option(GOOGLE_TEST "Build google tests" OFF)
 option(R_LIB "Build shared library for R package" OFF)
-set(GPU_COMPUTE_VER 35;50;52;60;61 CACHE STRING
-  "Space separated list of compute versions to be built against")
+
+# TODO
```

Comment:
@ruslo: FYI, I've tested linking `nccl` in xgboost builds locally with Hunter via `GIT_SUBMODULE` using:

```
xgboost_opts=(
    GOOGLE_TEST=ON
    USE_CUDA=ON
    CUDA_TOOLKIT_ROOT_DIR=/Developer/NVIDIA/CUDA-9.0
    XGBOOST_BUILD_SHARED=OFF
    R_LIB=OFF
    JVM_BINDINGS=OFF
)

polly.py --toolchain xcode --verbose --reconfig --config Debug --fwd ${xgboost_opts[@]} ${*}
```

Using CUDA's proprietary SYSTEM libs introduces a wrinkle.  I believe this typically needs some custom arguments like the `--fwd CUDA_TOOLKIT_ROOT_DIR=/Developer/NVIDIA/CUDA-9.0` to work on most platforms.  This doesn't currently fit with the hunter `./jenkins.py` test.  I don't have a multi-gpu rig for testing this anyway, so I've left nccl stuff stubbed out in xgboost for the initial release until the CUDA stuff can be resolved.

Reply:
> Using CUDA's proprietary SYSTEM libs introduces a wrinkle. I believe this typically needs some custom arguments like the --fwd CUDA_TOOLKIT_ROOT_DIR=/Developer/NVIDIA/CUDA-9.0 to work on most platforms

I haven't used it so I don't know what peculiarities it has. If we can download and build stuff without buying/registering license then it's not a problem for Hunter, we can add binary package. It will be user's responsibility to deal with runtime. The question here will be the size of download though.

The second option is to make a wrapper package that can find and import CUDA libraries. Will needed to set default version in `configs/default.cmake` and save version of CUDA in Config-ID to allow multiple non-conflict configurations.

Reply:
I'm going to "promote" the  CUDA discussion to the main hunter issues.  AFAIK, CUDA itself can be managed by hunter using "plain" downloads.

For example:

https://developer.nvidia.com/cuda-downloads?target_os=MacOSX&target_arch=x86_64&target_version=1012&target_type=dmgnetwork

There are other NVIDIA libs, such as `nccl v2.0` which require registration.

Managing these installations with hunter seems to make sense.

Reply:
Initial tests will have to be CPU only.